conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -182,22 +182,28 @@ public static List<Integer> newNonNullIntegerList(Iterable<Integer> iterable) {
         return integerList;
     }
 
-    // This method returns a list that can't handle nulls. Do not use this unless the nonNullCollections flag is set
+    /**
+     * This should only ever be called by a previously generated conjure internal implementation.
+     *
+     * @return a nonnull boxed list
+     * @deprecated This method is here for GCV back-compat purposes
+     */
+    @Deprecated
+    @DoNotCall
```

Comment:
let's not chance it with these annotations -- some projects override our defaults and get warnings from generated code. It's not great, but I don't want to block conjure-java rollout unnecessarily. We can recommend against these methods in docs, then deprecate/dnc in a month or two.